src/drivers/imx/sdma.c

```diff
@@ -265,6 +265,14 @@ static int sdma_upload_context(struct dma_chan_data *chan)
 			   sizeof(*pdata->ctx) / 4);
 }
 
+#if CONFIG_HAVE_SDMA_FIRMWARE
+static int sdma_load_firmware(struct dma *dma, void *buf, int addr, int size)
+{
+	return sdma_run_c0(dma->chan->dma, SDMA_CMD_C0_SET_PM,
+			   (uint32_t)buf, addr, size / 2);
```

Comment:
Maybe compiler finally change /2 to >>1, maybe not, I suggest SOF eliminate /2 /4 /8 operation and use >> instead.